MapboxNavigation/NavigationViewController.swift

```diff
@@ -200,6 +200,26 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
     
     var bottomViewController: ContainerViewController?
     
+    public typealias FloatingButtonsPositions = NavigationView.FloatingButtonsPositions
+    
+    open var floatingButtonsPosition: FloatingButtonsPositions? {
+        get {
+            return mapViewController?.floatingButtonsPosition
+        }
+        set {
+            mapViewController?.floatingButtonsPosition = newValue
+        }
+    }
+    
+    open var floatingButtons: [String:Button]? {
```

Comment:
This design requires each button to have a unique ID. In general, UIKit and SwiftUI operate under the assumption that views do not have unique IDs. Instead, application code normally holds onto pointers to specific view instances in memory. In other words, if the application wants to specify a mute button, it should refer to `NavigationView.muteButton` (though that property would need to be exposed publicly through NavigationViewController).

Reply:
Yes. The dictionary design has some problems related with the pointing. The array could allow the button to change the image and  the other property but still keeps its function. For example, by replacing the image and change the color, but this button still keeps the function of `mute` with pointing to the original `NavigationView.muteButton`  as the gif below:

```
        let tempButton = Button(frame: CGRect(x: 100, y: 100, width: 50, height: 50))
        tempButton.backgroundColor = UIColor.red
        if let tempFloatingButton = navigationViewController.floatingButtons?[1],
           let temp2 = navigationViewController.floatingButtons?[0] {
            tempFloatingButton.backgroundColor = UIColor.green
            tempFloatingButton.setImage(temp2.currentImage, for: .normal)
            navigationViewController.floatingButtons = [tempFloatingButton, tempButton]
        }
```
![customizedMute](https://user-images.githubusercontent.com/48976398/102151500-e7e31e00-3e27-11eb-9cce-635297a5f303.gif)